### ISSUE TITLE
Backend does not respect disabled IPv6 in profile

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-backend#1039]
If  IPv6 is set to false in the default profile (/usr/local/share/perl/5.26.1/auto/share/dist/Zonemaster-Engine/profile.json on Ubuntu) the setting is respected by both Backend and CLI. If we instead create a new profile file and set IPv6 off in that file and use it, CLI will respect that IPv6 is turned off, but Backend will not.

Copy the default profile to "/etc/zonemaster/profile-changed.json" and introduce the following changes:
```
--- profile.json	2022-06-12 12:00:49.729268150 +0200
+++ profile-changed.json	2022-06-12 12:00:43.165354222 +0200
@@ -8,7 +8,7 @@
     },
     "net" : {
         "ipv4" : true,
-        "ipv6" : true
+        "ipv6" : false
     },
     "no_network" : false,
     "resolver" : {
@@ -57,11 +57,11 @@
             "HAS_A_RECORDS" : "ERROR",
             "HAS_NAMESERVER_NO_WWW_A_TEST" : "INFO",
             "HAS_NAMESERVERS" : "INFO",
-            "HAS_PARENT" : "INFO",
-            "IPV4_DISABLED" : "INFO",
-            "IPV4_ENABLED" : "DEBUG",
-            "IPV6_DISABLED" : "INFO",
-            "IPV6_ENABLED" : "DEBUG",
+            "HAS_PARENT" : "ERROR",
+            "IPV4_DISABLED" : "NOTICE",
+            "IPV4_ENABLED" : "NOTICE",
+            "IPV6_DISABLED" : "NOTICE",
+            "IPV6_ENABLED" : "NOTICE",
             "NO_GLUE_PREVENTS_NAMESERVER_TESTS" : "CRITICAL",
             "NO_PARENT" : "CRITICAL",
             "NS_FAILED" : "ERROR",
```
Update "backend_config.ini" with the following:
```
[PUBLIC PROFILES]
default=/etc/zonemaster/profile-changed.json
```
Restart Backend and start a test with `zmb start_domain_test` (not with `zmtest` since it always turn IPv6 on, #1040) and fetch the result with `zmb get_test_results`. 

Note that the new profile is used because in the results you will find changed levels, but IPv6 is still enabled.
```
      {
        "message": "Parent domain 'fr' was found for the tested domain.\n",
        "level": "ERROR",
        "module": "BASIC"
      },
      {
        "message": "IPv4 is enabled,  can send \"NS\" query to ns1.nic.fr/192.134.4.1.\n",
        "level": "NOTICE",
        "module": "BASIC"
      },
      {
        "level": "NOTICE",
        "message": "IPv6 is enabled,  can send \"SOA\" query to ns1.nic.fr/2001:67c:2218:2::4:1.\n",
        "module": "BASIC"
      }
```
Use the profile with zonemaster-cli and you will see that IPv6 is turned off:
```
zonemaster-cli --profile /etc/zonemaster/profile-changed.json zonemaster.net --test basic
Loading profile from /etc/zonemaster/profile-changed.json.

Seconds Level     Message
======= ========= =======
   0.49 ERROR     Parent domain 'net' was found for the tested domain.
   2.15 NOTICE    IPv4 is enabled, can send "SOA" query to ns2.nic.fr/192.93.0.4.
   2.15 NOTICE    IPv4 is enabled, can send "NS" query to ns2.nic.fr/192.93.0.4.
   2.16 NOTICE    IPv6 is disabled, not sending "SOA" query to ns2.nic.fr/2001:660:3005:1::1:2.
   2.16 NOTICE    IPv6 is disabled, not sending "NS" query to ns2.nic.fr/2001:660:3005:1::1:2.
   2.16 NOTICE    IPv4 is enabled, can send "SOA" query to nsa.dnsnode.net/194.58.192.46.
   2.16 NOTICE    IPv4 is enabled, can send "NS" query to nsa.dnsnode.net/194.58.192.46.
   2.19 NOTICE    IPv6 is disabled, not sending "SOA" query to nsa.dnsnode.net/2a01:3f1:46::53.
   2.19 NOTICE    IPv6 is disabled, not sending "NS" query to nsa.dnsnode.net/2a01:3f1:46::53.
(...)
```

**Comment by matsduf** [comment]
If the parameters "ipv4" or "ipv6" are unset then RPC API will give a value to them based on the value in the default profile. If the test has started with a "profile" parameter, that profile is not loaded by RPC API.

When TestAgent picks the test up, it will load the profile from the parameter, and after that let the parameters "ipv4" and "ipv6" overwrite what is set in the profile.

The profile from the "profile" parameter must be loaded by RPC API before unset "ipv4" or "ipv6" are set with values.

**Comment by matsduf** [comment]
Resolved by #1046
